contracts/examples/OracleCreator.sol

```diff
@@ -0,0 +1,183 @@
+pragma solidity =0.6.6;
+
+import './../libraries/TransferHelper.sol';
+import './../libraries/DXswapOracleLibrary.sol';
+import './../libraries/DXswapLibrary.sol';
+import './../libraries/SafeMath.sol';
+
+contract OracleCreator {    
+    using FixedPoint for *;
+    using SafeMath for uint256;
+
+    event OracleCreated(
+        uint256 _oracleId,
+        address _pair,
+        uint256 _windowSize,
+        uint8 _granularity,
+        uint256 _periodSize,
+        uint256 _executionBountyWei
+    );
+
+    event OracleExecuted(
+        uint256 _oracleId
+    );
+
+    struct Oracle{
+      uint256 windowSize;
+      address pair;
+      uint8 granularity;
+      uint256 periodSize;
+      uint256 executionBountyWei;
+      uint256 finishedTimestamp;
+      bool observationFinished;
+    }
+
+    struct Observation {
+      uint256 timestamp;
+      uint256 price0Cumulative;
+      uint256 price1Cumulative;
+    }
+
+    address public immutable DXswapRelayer;
+    mapping(uint256 => Oracle) public oracles;
+    uint256 public oraclesIndex;
+    mapping (uint256 => Observation[]) public oracleObservations;
+    
+    constructor(
+        address payable _DXswapRelayer
+    ) public {
+        DXswapRelayer = _DXswapRelayer;
+    }
+
+    function createOracle(
+        uint256 windowSize,
+        address pair,
+        uint8 granularity,
+        uint256 executionBountyWei
+    ) public payable returns (uint256 oracleId) {
+        uint256 periodSize;
+        require(msg.sender == DXswapRelayer, 'DXliquidityRelay: CALLER_NOT_RELAYER');
+        require(granularity > 1, 'DXliquidityRelay: GRANULARITY');
+        require(
+            (periodSize = windowSize / granularity) * granularity == windowSize,
```

Comment:
periodSize = windowSize / granularity) * granularity == windowSize

periodSize is assigned before the requires and it is always zero, It would be best to check the condition against zero and then after all the requires assign the periodSize uint.

Reply:
@AugustoL the require will make sure it's evenly divisible and assigns it to the state variable in the same line